src/components/Api.js

```diff
@@ -0,0 +1,78 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleResponse(res) {
+    if (!res.ok) {
+      return Promise.reject(
+        new Error(`Error ${res.status}: ${res.statusText}`)
+      );
+    }
+    return res.json();
+  }
+
+  // **Cards** //
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "GET",
+      headers: this._headers,
+    }).then(this._handleResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._handleResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.